pytest_tests/tests/object/test_object_api.py

```diff
@@ -592,6 +592,52 @@ def test_object_get_range(self, request: FixtureRequest, storage_objects: list[S
                         == range_content
                     ), f"Expected range content to match {range_cut} slice of file payload"
 
+    @allure.title("Validate native object API get_range for a complex object")
+    @pytest.mark.grpc_api
+    def test_object_get_range_complex(self, default_wallet: NodeWallet, container: str, complex_object_size: int):
+        """
+        Validate get_range for object by native gRPC API for a complex object
+        """
+        file_path = generate_file(complex_object_size)
+        oid = put_object_to_random_node(
+            default_wallet.path,
+            file_path,
+            container,
+            shell=self.shell,
+            neofs_env=self.neofs_env,
+        )
+
+        file_ranges_to_test = []
+
+        parts = get_object_chunks(default_wallet.path, container, oid, self.shell, self.neofs_env)
+
+        # range is inside one child
+        file_ranges_to_test.append((0, parts[0][1] - 1))
+        # range matches child
+        file_ranges_to_test.append((parts[0][1], parts[1][1]))
+        # range requires more than one child (it includes OR does not include the first and the last children)
+        file_ranges_to_test.append((0, parts[0][1] + parts[1][1] - 1))
+        # range requires more than two children (it includes OR does not include the first and the last children)
+        file_ranges_to_test.append((0, complex_object_size - 1))
+        file_ranges_to_test.append((parts[0][1] + 1, complex_object_size - parts[0][1] - parts[-1][1] - 1))
```

Comment:
looks working correctly but really relies on the fact that there will always be more than two parts, not sure if it is error-proof

Reply:
With the current settings there always will be more than two parts. If in the future we will update something, we will update this accordingly. Making an error-proof test usually is a waste of time.